### PR TITLE
Please anyone consults my question about Segwit

### PR DESCRIPTION
**Pull request by youngmtool**
1. 
  At this sentence:
  Before Segwit the transaction signature was used in the calculation of the transaction id.
  
    
  What's "the transaction signature" meaning?
  
  Is it a ScriptSig in a input of a transaction?
  
  As to the above sentence, can I say like this:
  (According to an illustration, it looks like, since TxId is generated by inputs and outputs,)
  Before Sewgit the transaction signature(ScriptSig) was used in the calculation of the transaction id along with other values of input and entire of output.
  
  
  
  2. 
  At this sentence:
  Linear sig hash scaling: Signing a transaction used to require hashing the whole transaction for every input. This was a potential DDoS vector attack for large transactions.
  
  Considering this part:
  Signing a transaction used to require hashing the whole transaction for every input.
  
  Signing a transaction means generating transactionID?
  
  If generating transactionID requires hashing the "whole transaction", the "whole transaction" means block header(previous transactionID, metadata, and so on) + block body(inputs, outputs) or only body part?
  
  3.
  On an illustration, on the left one, Signature1 and Signature2 mean the ScriptSig? Is it possible to exsist multiple ScriptSigs in one input?
  
  4.
  On an illustration, on the right one, even though the Signature1 and Signature2 are located out of both inputs and outputs, Signature1 and Signature2 values are located in input, according to this sentence:
  Basically, Segregated Witness moves the proof of ownership from the scriptSig part of the transaction to a new part called the witness of the input.
  
  Can I say that the reason for the location of Signatures on an illustration is just to show after Segwit TxId is generated by inputs and outputs without Signature(previously ScriptSig value) which is after the Segwit, the value of witness in a input?

### CODE REVIEW

**Comment by nopara73** [comment]
> Is it a ScriptSig in a input of a transaction?

It's the scriptsig of the input (not input transaction). And with segwit goes to witness and scriptsig remains empty.

> Signing a transaction means generating transactionID?

Think about it like this:  
The hash of the transaction is the transaction id. This hashing includes the scriptsig, but does not include the witness. So if the scriptsig is empty, then no matter how miners malleate the witness the txid won't change.

> "whole transaction" means block header(previous transactionID, metadata, and so on)

A transaction is just a list of inputs (pointers to unspent transaction outputs plus signatures) and outputs (scriptpubkeys and values). And some minimal data. You can see it when you ToString() a transaction.  

> On an illustration, on the left one, Signature1 and Signature2 mean the ScriptSig? 

Yes.  

> Is it possible to exsist multiple ScriptSigs in one input?

No.

I'm kinda confused by your fourth point.

**Comment by youngmtool** [comment]
Thanks @nopara73 .

Got clear with many concepts.

But of your comment of this:
I'm kinda confused by your fourth point.

Are you confused by the way of sentence expression of my question or by the corresponding contents themselves?

Since I understood something a little, I want to mention again what I wanted to ask.

At this page:
https://github.com/youngmtool/ProgrammingBlockchain/blob/42a40eb70a8882ac4db7b5165b6479f60f7f5c58/other_types_of_ownership/p2wpkh_pay_to_witness_public_key_hash.md


For me, the illustration is a little confusing.

On the left one, Signature1 and Signature2 mean each ScriptSig in each input before Segwit.
But on the right one, Signature1 and Signature2 should mean each witness.
Moreover, Signature1 and Signature2 meaning each witness should be located in each input physically.
The reason that Signature1 and Signature2 meaning each witness are out of both inputs and outputs is for the purpose of showing that TxID is generated without Signature1 and Signature2 meaning each witness after Segwit.

Is there any wrong concept from above understanding?

**Comment by youngmtool** [comment]
And please see here:
https://github.com/ProgrammingBlockchain/ProgrammingBlockchain/pull/81

I got this message, conflict:
This branch has conflicts that must be resolved.

Why did it happen?

I tried "Resolve conflicts", and then I got this message:
Merge branch 'master' into master.

**Comment by nopara73** [comment]
> On the left one, Signature1 and Signature2 mean each ScriptSig in each input before Segwit.
But on the right one, Signature1 and Signature2 should mean each witness.  

Yes, but depicting it that way would be more confusing. Note, the goal of the illustration is to show: "hey, before we had the signatures in inputs, now we have out of the transaction".  Adding the scriptsig and witness terminology to the illustration would confuse some.  

> Moreover, Signature1 and Signature2 meaning each witness should be located in each input physically.
> The reason that Signature1 and Signature2 meaning each witness are out of both inputs and outputs is for the purpose of showing that TxID is generated without Signature1 and Signature2 meaning each witness after Segwit.

Yes.

> This branch has conflicts that must be resolved.

Resolving conflicts are always difficult. Maybe it's a good time to learn more about Git:)

**Comment by youngmtool** [comment]
Yes, Git. Okay, thanks.

**Comment by nopara73** [comment]
I close this. I think you accidentally opened a PR instead of the issue.
